modules/aws-waf/main.tf

```diff
@@ -0,0 +1,50 @@
+# ------------------------------------
+# WEB APPLICATION FIRWEWALL
+# ------------------------------------
+
+resource "aws_waf_ipset" "ipset" {
+  count = length(var.allowed_ips) > 0 ? 1 : 0
+  name  = "${var.name}-ipset"
+  dynamic "ip_set_descriptors" {
+    iterator = ip
+    for_each = var.allowed_ips
+    content {
+      type  = "IPV4"
+      value = ip.value
+    }
+  }
+}
+
+resource "aws_waf_rule" "wafrule" {
+  count       = length(var.allowed_ips) > 0 ? 1 : 0
+  depends_on  = [aws_waf_ipset.ipset]
```

Comment:
Is `depends_on` necessary here? Terraform should be able to automagically detect dependencies 🤔
I guess that this may be necessary with nested blocks though

Reply:
This was copied from terraform.io example, seems to work without this one so I'll remove it